adapters/smartyads/smartyads.go

```diff
@@ -0,0 +1,209 @@
+package smartyads
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"text/template"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+// Implements Bidder interface.
+type SmartyAdsAdapter struct {
+	endpoint template.Template
+}
+
+func NewSmartyAdsBidder(endpointTemplate string) *SmartyAdsAdapter {
+	template, err := template.New("endpointTemplate").Parse(endpointTemplate)
+	if err != nil {
+		return nil
+	}
+	return &SmartyAdsAdapter{endpoint: *template}
+}
+
+func (a *SmartyAdsAdapter) CheckHasImps(request *openrtb.BidRequest) error {
```

Comment:
This check isn't necessary. The "core" code will not call an adapter if there are no imps.